org-clock-csv.el

```diff
@@ -243,6 +242,19 @@ for use in batch mode."
             entries))
     (switch-to-buffer buffer)))
 
+;;;###autoload
+(defun org-clock-export-to-csv-file (csv-file &optional infile)
+  "Export clock entries from INFILE to CSV-FILE.
+
+See `org-clock-csv' function for INFILE usage. If INFILE is nil,
+`org-agenda-files' is used as clock source."
+  (interactive "FCSV file: ")
+  (org-clock-csv infile)
+  (with-current-buffer "*clock-entries-csv*"
+    (rename-buffer csv-file)
+    (set-visited-file-name csv-file)
+    (save-buffer)))
```

Comment:
Instead of these three comands use `(write-file FILENAME &optional CONFIRM)` or even better `(write-region START END FILENAME &optional APPEND VISIT LOCKNAME MUSTBENEW)` (visiting files is not great as it locks things and may break concurrent access).

Reply:
There's also another option which might be the simplest

```elisp
(with-temp-file outfile
  (insert
   (with-current-buffer (org-clock-csv infile)
     (buffer-string))))
```